### PR TITLE
Add reactizing a complex backbone view post

### PR DESCRIPTION
**Pull request by baseten**
Post on Reactizing the Custom Fields Modal including info on Backbone Testing Library and methodology of adding test coverage first.

### CODE REVIEW

**Comment by baseten** [comment]
@craigmdennis it might be nice if the callout section rendered a little differently (more than just a margin)?

File: _posts/2020-12-23-reactizing-a-complex-backbone-view/index.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+pattern described above will help with this, but you'll also need some sort of utility to 
+render React Components from Backbone. We have a `BaseView` Backbone class to help 
+handle this, with a method called `renderComponent`, which takes the component (as JSX 
+with props) as the first argument, and a selector pointing to a child DOM element as the 
+second. Apart from keeping track of rendered components for later cleanup, it ultimately 
+calls `ReactDOM.render` - inserting your React controlled component into the DOM beneath
+the Backbone controlled parent View.
+
+Once integration is done, don't forget to manually test! If you have E2E tests even 
+better, but you're unit tests obviously aren't going to catch any integration issues. If
```
Comment (lukasmlady) [inline _posts/2020-12-23-reactizing-a-complex-backbone-view/index.md:196]: typo: should be `your`

File: _posts/2020-12-23-reactizing-a-complex-backbone-view/index.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+as much as possible.
+
+# Summary (TLDR)
+
+1. Audit the UI **and** code.
+2. Ensure the existing Backbone View has "full" test coverage.
+3. Port tests over to a new empty React Component.
+4. Build out the new React component using TDD, **don't be tempted to add new features yet**.
+5. Replace the Backbone View with the new React Component and do full E2E/manual tests.
+6. Delete the Backbone View
```
Comment (lukasmlady) [inline _posts/2020-12-23-reactizing-a-complex-backbone-view/index.md:38]: missing a dot here and point 7 at the end of the sentence

File: _posts/2020-12-23-reactizing-a-complex-backbone-view/index.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+When thinking about this tradeoff and trying to decide when the time is right
+to invest in a significant refactor, I'm always reminded of Joel Spolsky's [Things You 
+Should Never Do](https://www.joelonsoftware.com/2000/04/06/things-you-should-never-do-part-i/).
+In this case **we really do want** to throw our code away, but we want to do it in a
+controlled and manageable way, keeping the years of effort and bugfixes that have gone
+into it already. Below is a breakdown of the process we've developed to mitigate problems 
+as much as possible.
+
+# Summary (TLDR)
```
Comment (lukasmlady) [inline _posts/2020-12-23-reactizing-a-complex-backbone-view/index.md:31]: nit: `TL;DR` feels more familiar

File: _posts/2020-12-23-reactizing-a-complex-backbone-view/index.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+after all we only care about manipulating the DOM not how it's output, but `Cypress` tests 
+are kinda slow, and it means we'd need to instrument the whole app, make sure we had seed 
+data and navigated to the right page, etc.
+
+[The paradigm behind `react-testing-library`](https://testing-library.com/docs/react-testing-library/intro/#the-problem)
+is also to test the DOM, not the implementation! Can that help? Well,
+[`react-testing-library`](https://github.com/testing-library/react-testing-library) is a
+wrapper around [`dom-testing-library`](https://github.com/testing-library/dom-testing-library),
+and really it's all just about interacting with the DOM after rendering. We'd already
+built ourselves `backbone-testing-library` to be able to write our tests in a similar
```
Comment (lukasmlady) [inline _posts/2020-12-23-reactizing-a-complex-backbone-view/index.md:100]: `backbone-testing-library` could be a link as the react-* and dom-* ones are also links

**Review by lukasmlady — APPROVED** [review]
Overall like the spirit of the post, only left a couple of minor comments. Thanks for putting this together!

File: _posts/2020-12-23-reactizing-a-complex-backbone-view/index.md (hunk 1 above)
Comment (baseten) [reply _posts/2020-12-23-reactizing-a-complex-backbone-view/index.md]: resolved in 4289f62ef80bb4de33d4817d5f92eb74884ca9e9

File: _posts/2020-12-23-reactizing-a-complex-backbone-view/index.md (hunk 2 above)
Comment (baseten) [reply _posts/2020-12-23-reactizing-a-complex-backbone-view/index.md]: resolved in 4289f62ef80bb4de33d4817d5f92eb74884ca9e9

File: _posts/2020-12-23-reactizing-a-complex-backbone-view/index.md (hunk 3 above)
Comment (baseten) [reply _posts/2020-12-23-reactizing-a-complex-backbone-view/index.md]: resolved in 4289f62ef80bb4de33d4817d5f92eb74884ca9e9

File: _posts/2020-12-23-reactizing-a-complex-backbone-view/index.md (hunk 4 above)
Comment (baseten) [reply _posts/2020-12-23-reactizing-a-complex-backbone-view/index.md]: resolved in 4289f62ef80bb4de33d4817d5f92eb74884ca9e9

**Review by lukasmlady — APPROVED** [review]

File: _posts/2020-12-23-reactizing-a-complex-backbone-view/index.md [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+title: A Case Study in "Reactizing" a Complex Backbone View
+date: 2020-12-23
+permalink: /posts/reactizing-a-complex-backbone-view
+author: Alex Prokop
+thumbnail: ''
+metaDescription: ''
+tags: [Engineering, Open Source, Testing, Javascript, React, Backbone]
+---
+
+As part of our new "Custom Activities" feature, it became apparent the changes required
```
Comment (philfreo) [inline _posts/2020-12-23-reactizing-a-complex-backbone-view/index.md:11]: should link this to the main blog post or FAQ about the feature

**Review by philfreo — COMMENTED** [review]
good post!

File: _posts/2020-12-23-reactizing-a-complex-backbone-view/index.md (hunk 5 above)
Comment (baseten) [reply _posts/2020-12-23-reactizing-a-complex-backbone-view/index.md:11]: Done in https://github.com/closeio/making.close.com/pull/50

**Pull request merged**